tests/playbooks/roles/install-csi-cinder/tasks/main.yaml

```diff
@@ -0,0 +1,206 @@
+              - name: openstack-cloud-controller-manager
+                image: docker.io/k8scloudprovider/openstack-cloud-controller-manager:latest
+                args:
+                  - /bin/openstack-cloud-controller-manager
+                  - --v=4
+                  - --cloud-config=/etc/config/cloud.conf
+                  - --cloud-provider=openstack
+                  - --use-service-account-credentials=true
+                  - --bind-address=127.0.0.1
+                volumeMounts:
+                  - mountPath: /etc/config
+                    name: cloud-config-volume
+                    readOnly: true
+            hostNetwork: true
+            volumes:
+            - name: cloud-config-volume
+              secret:
+                secretName: cloud-config
+      EOF
+
+- name: Wait for openstack-cloud-controller-manager up and running
+  shell:
+    executable: /bin/bash
+    cmd: |
+      kubectl -n kube-system get pod | grep openstack-cloud-controller-manager | grep Running
+  register: check_occm
+  until: check_occm.rc == 0
+  retries: 24
+  delay: 5
+  ignore_errors: yes
+
+- name: Deploy cinder-csi-plugin
+  shell:
+    executable: /bin/bash
+    cmd: |
+      cd $GOPATH/src/k8s.io/cloud-provider-openstack
+      kubectl apply -f manifests/cinder-csi-plugin
+
+- name: Wait for csi-cinder-controllerplugin up and running
+  shell:
+    executable: /bin/bash
+    cmd: |
+      kubectl -n kube-system get pod | grep csi-cinder-controllerplugin  | grep Running
+  register: check_csi
+  until: check_csi.rc == 0
+  retries: 24
+  delay: 5
+
+- name: Wait for csi-cinder-nodeplugin up and running
+  shell:
+    executable: /bin/bash
+    cmd: |
+      kubectl -n kube-system get pod | grep csi-cinder-nodeplugin  | grep Running
+  register: check_csi
+  until: check_csi.rc == 0
+  retries: 24
+  delay: 5
+
+- name: Gather additional evidence if csi-cinder-plugin failed to come up
+  when: check_csi.failed
+  block:
+    - name: Describe failed csi-cinder-plugin
+      shell:
+        executable: /bin/bash
+        cmd: |
+          kubectl -n kube-system describe deployment
+      register: describe_csi_cinder
+      changed_when: false
+
+    - name: Log failed csi-cinder-plugin deployment
+      debug:
+        var: describe_csi.stdout_lines
+
+    - name: &failmsg Stop due to prior failure of csi-cinder-plugin
+      fail:
+        msg: *failmsg
+
+- name: Run functional tests for csi-cinder-plugin
+  shell:
+    executable: /bin/bash
+    cmd: |
+      set -x
+      set -e
+      set -o pipefail
+
+      cd $GOPATH/src/k8s.io/cloud-provider-openstack
+
+      # GATEWAY_IP is the default value in devstack
+      GATEWAY_IP=172.24.5.1 \
+      OS_RC={{ devstack_workdir }}/openrc \
+      go test -v ./cmd/tests/cinder_csi_e2e_suite_test.go -ginkgo.v -ginkgo.progress -ginkgo.skip="\[Disruptive\]" -ginkgo.focus="\[cinder-csi-e2e\]" -ginkgo.noColor -timeout=0 -report-dir='{{ k8s_log_dir }}' | tee "{{ k8s_log_dir }}/e2e.log"
```

Comment:
We could use `ARTIFACTS="${ARTIFACTS:-${PWD}/_artifacts}"` to move the log file to a dedicated place so it's easy to check out.